libqtile/state.py

```diff
@@ -45,6 +53,39 @@ def apply(self, qtile):
             Rearrange the windows in the specified Qtile object according to
             this QtileState.
         """
+        try:
+            for group in qtile.groups:
+                x = []
+                for i in self.focusHistory[group.name]:
+                    x.append(qtile.windowMap[i.wid])
+                group.focusHistory = x
+                for layout in group.layouts:
+                    d = self.layoutMap[group.name][layout.name]
+                    d.group = layout.group
+                    members = dir(d)
+                    for member in members:
```

Comment:
All layouts are derived classes of the base layout class and they have multple attributes refering to `window` class.
If you were to see [Max Layout]( https://github.com/qtile/qtile/blob/develop/libqtile/layout/max.py) it has multiple attributes that refer to objects of window class like `clients` , `current` etc. Similarly other layouts have different attributes with different names that refer to `window` class objects.